### ISSUE TITLE
[Backport] k8s v1.16.0-beta.1 cluster provisioning fails with aws cloud provider

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#23544]
[backport] rancher/rancher#22814

**Comment by sangeethah** [comment]
https://github.com/rancher/rancher/pull/23550

**Comment by jiaqiluo** [comment]
The bug is reproduced in Rancher: v2.3.1, the steps are the same as the issue provides.

The RKE cluster is configured with a could provider (AWS).

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher:v2.3-head 
steps:
- run Rancher single install
- add an RKE cluster with cloud provider AWS

results:
- the `Failed to retrieve node info: ` error is not shown in `kube-proxy`'s log anymore.
- instead, the following message shows continually  
```
I1018 23:38:34.708899   14031 proxier.go:798] Not using `--random-fully` in the MASQUERADE rule for iptables because the local version of iptables does not support it
```

**Comment by jiaqiluo** [comment]
As discussed with @superseb, the bug is fixed. 
As to the verbose messages in the log, a new issue is made to track it: https://github.com/rancher/rancher/issues/23552
